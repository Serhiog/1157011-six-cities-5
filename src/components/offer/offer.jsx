import React, {useEffect} from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import MainPage from "../main-page/main-page";
import ReviewList from "../review-list/review-list";
import {fetchHotelReviews} from "../../store/api-actions";
import Map from "../map/map";
import NearbyOffersList from "../nearby-offers-list/nearby-offers-list";
import {connect} from "react-redux";
import {getCurrentOffer, getNearbyOffers} from "../../store/offers/selectors";
import {PropTypes4Offer} from "../../propConsts";

const Offer = ({noLogged = true, actualOffer, offers, getReviews}) => {

  useEffect(() => {
    getReviews(actualOffer.id);
  }, []);

  return (
    <div className="page">
      <header className="header">
        <div className="container">
          <div className="header__wrapper">
            <div className="header__left">
              <Link className="header__logo-link" to={`/`}>
                <img
                  className="header__logo"
                  src="/img/logo.svg"
                  alt="6 cities logo"
                  width={81}
                  height={41}
                />
              </Link>
            </div>
            <nav className="header__nav">
              <ul className="header__nav-list">
                <li className="header__nav-item user">
                  <Link
                    className="header__nav-link header__nav-link--profile"
                    to={<MainPage />}
                  >
                    <div className="header__avatar-wrapper user__avatar-wrapper"></div>

                    {noLogged ? (
                      <span className="header__login">Sign in</span>
                    ) : (
                      <span className="header__user-name user__name">
                        Oliver.conner@gmail.com
                      </span>
                    )}
                  </Link>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {actualOffer.images.map((photo, i) => {
                return (
                  <div key={i} className="property__image-wrapper">
                    <img
                      className="property__image"
                      src={photo}
                      alt={actualOffer.description}
                    />
                  </div>
                );
              })}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {actualOffer.is_premium ? (
                <div className="property__mark">
                  <span>Premium</span>
                </div>
              ) : (
                ``
              )}
              <div className="property__name-wrapper">
                <h1 className="property__name">{actualOffer.description}</h1>
                <button
                  className="property__bookmark-button button"
                  type="button"
                >
                  <svg
                    className="property__bookmark-icon"
                    width={31}
                    height={33}
                  >
                    <use xlinkHref="#icon-bookmark" />
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: actualOffer.rating * 10}} />
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">
                  {actualOffer.rating * 10}
                </span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {actualOffer.type}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {actualOffer.bedrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {actualOffer.max_adults} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">â‚¬{actualOffer.price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {actualOffer.goods.map((feature, i) => {
                    return (
                      <li key={i} className="property__inside-item">
                        {feature}
                      </li>
                    );
                  })}
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                    <img
                      className="property__avatar user__avatar"
                      src={actualOffer.host.avatar_url}
                      width={74}
                      height={74}
                      alt="Host avatar"
                    />
                  </div>
                  <span className="property__user-name">
                    {actualOffer.host.name}
                  </span>
                </div>
                <div className="property__description">
                  <p className="property__text">
                    A quiet cozy and picturesque that hides behind a a river by
                    the unique lightness of Amsterdam. The building is green and
                    from 18th century.
                  </p>
                  <p className="property__text">
                    An independent House, strategically located between Rembrand
                    Square and National Opera, but where the bustle of the city
                    comes to rest in this alley flowery and colorful.
                  </p>
                </div>
              </div>
              <ReviewList offer={actualOffer} />
            </div>
          </div>
          <section className="property__map map">
            <Map offers={offers} />
          </section>
        </section>
        <div className="container">
          <NearbyOffersList offers={offers} actualOffer={actualOffer} />
        </div>
      </main>
    </div>
  );
};

Offer.propTypes = {
  noLogged: PropTypes.bool,
  actualOffer: PropTypes.shape(PropTypes4Offer),
  offers: PropTypes.arrayOf(PropTypes.shape(PropTypes4Offer)),
  getReviews: PropTypes.func.isRequired
};


const mapToStateProps = (state) => ({
  actualOffer: getCurrentOffer(state),
  offers: getNearbyOffers(state),
});

const mapDispatchToProps = (dispatch) => ({
  getReviews(offerId) {
    dispatch(fetchHotelReviews(offerId));
  },
});

export {Offer};
export default connect(mapToStateProps, mapDispatchToProps)(Offer);
