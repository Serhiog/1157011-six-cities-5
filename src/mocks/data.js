
export const cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

export const offers = [
  {
    id: 0,
    isFavorite: false,
    isPremium: false,
    features: {
      type: `hotel`,
      badrooms: 1,
      adults: 2,
    },
    price: 100,
    title: `TITLE`,
    rating: 3.4,
    things: [
      `Wifi`,
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Own bathroom`,
      `Hair-dryer`,
      `AC`,
    ],
    previewPictures: `img/apartment-small-01.jpg`,
    pictures: [
      `img/apartment-small-03.jpg`,
      `img/apartment-small-04.jpg`,
      `img/room-small.jpg`,
      `img/studio-photos.jpg`,
    ],
    description: `The delusion emphasizes intelligent common sense, although the opposite is accepted in the official language.`,
    owner: {
      id: 0,
      name: `Spider Man`,
      super: true,
      picture: `img/avatar-angelina.jpg`,
    },
    detailsMapZoom: 12,
    coordinates: [52.3909553943508, 4.85309666406198],
    mapZoom: 10,
    city: `Amsterdam`,
    cityCoordinates: [52.38333, 4.9],
  },
  {
    id: 2,
    isFavorite: true,
    isPremium: false,
    features: {
      type: `hotel`,
      badrooms: 2,
      adults: 4,
    },
    price: 200,
    title: `TITLE - 2`,
    rating: 4.5,
    things: [
      `Wifi`,
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Own bathroom`,
      `Hair-dryer`,
      `AC`,
    ],
    previewPictures: `img/apartment-small-01.jpg`,
    pictures: [
      `img/apartment-small-03.jpg`,
      `img/apartment-small-04.jpg`,
      `img/room-small.jpg`,
      `img/studio-photos.jpg`,
    ],
    description: `The delusion emphasizes intelligent common sense, although the opposite is accepted in the official language.`,
    owner: {
      id: 1,
      name: `Iron Man`,
      super: false,
      picture: `img/avatar-angelina.jpg`,
    },
    detailsMapZoom: 12,
    coordinates: [52.3909553943508, 4.8530966640619],
    mapZoom: 10,
    city: `Amsterdam`,
    cityCoordinates: [52.38333, 4.9],
  },
  {
    id: 3,
    isFavorite: false,
    isPremium: true,
    features: {
      type: `private room`,
      badrooms: 1,
      adults: 6,
    },
    price: 300,
    title: `TITLE - 3`,
    rating: 4.9,
    things: [
      `Wifi`,
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Own bathroom`,
      `Hair-dryer`,
      `AC`,
    ],
    previewPictures: `img/apartment-small-01.jpg`,
    pictures: [
      `img/apartment-small-03.jpg`,
      `img/apartment-small-04.jpg`,
      `img/room-small.jpg`,
      `img/studio-photos.jpg`,
    ],
    description: `The delusion emphasizes intelligent common sense, although the opposite is accepted in the official language.`,
    owner: {
      id: 3,
      name: `Billi Willi`,
      super: true,
      picture: `img/avatar-angelina.jpg`,
    },
    detailsMapZoom: 12,
    coordinates: [52.390955394351, 4.85309666406201],
    mapZoom: 10,
    city: `Amsterdam`,
    cityCoordinates: [52.38333, 4.9],
  },
  {
    id: 4,
    isFavorite: true,
    isPremium: true,
    features: {
      type: `house`,
      badrooms: 6,
      adults: 3,
    },
    price: 400,
    title: `TITLE - 4`,
    rating: 2.1,
    things: [
      `Wifi`,
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Own bathroom`,
      `Hair-dryer`,
      `AC`,
    ],
    previewPictures: `img/apartment-small-01.jpg`,
    pictures: [
      `img/apartment-small-03.jpg`,
      `img/apartment-small-04.jpg`,
      `img/room-small.jpg`,
      `img/studio-photos.jpg`,
    ],
    description: `The delusion emphasizes intelligent common sense, although the opposite is accepted in the official language.`,
    owner: {
      id: 4,
      name: `Agent 007`,
      super: true,
      picture: `img/avatar-angelina.jpg`,
    },
    detailsMapZoom: 12,
    coordinates: [52.3909553943515, 4.85309666406196],
    mapZoom: 10,
    city: `Amsterdam`,
    cityCoordinates: [52.38333, 4.9],
  },
];

export const reviews = [
  {
    id: 1,
    picture: `img/avatar-max.jpg`,
    name: `Spider Man`,
    rating: 2,
    review: `She carefully took a crumpled, half-torn ten-dollar bill from her purse and handed it to him. He shoved a white`,
    date: `2020-05-08T14:13:56.569Z`,
    super: true,
  },
  {
    id: 2,
    picture: `img/avatar-max.jpg`,
    name: `Aqua Man`,
    rating: 5,
    review: `Be Happy! too! She carefully took a crumpled, half-torn ten-dollar bill from her purse and handed it to him. He shoved a white`,
    date: `2020-06-08T14:30:56.569Z`,
    super: true,
  },
];

export const serverOffers = [
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: `Amsterdam`,
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
    ],
    host: {
      avatar_url: `img/1.png`,
      id: 3,
      is_pro: true,
      name: `Angelina`,
    },
    id: 1,
    images: [`img/1.png`, `img/2.png`],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8,
    },
    max_adults: 4,
    preview_image: `img/1.png`,
    price: 120,
    rating: 4.8,
    title: `Beautiful & luxurious studio at great location`,
    type: `apartment`,
  },
  {
    bedrooms: 4,
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: `Amsterdam`,
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [
      `Heating`,
      `Kitchen`,
      `Cable TV`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
    ],
    host: {
      avatar_url: `img/1.png`,
      id: 3,
      is_pro: true,
      name: `Angelina`,
    },
    id: 2,
    images: [`img/1.png`, `img/2.png`],
    is_favorite: false,
    is_premium: true,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8,
    },
    max_adults: 4,
    preview_image: `img/1.png`,
    price: 100,
    rating: 4.8,
    title: `Beautiful & luxurious studio at great location`,
    type: `apartment`,
  },
];

export const serverReviews = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    user: {
      avatar_url: `img/1.png`,
      id: 4,
      is_pro: false,
      name: `Max`,
    },
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 2,
    rating: 4,
    user: {
      avatar_url: `img/1.png`,
      id: 4,
      is_pro: false,
      name: `Max`,
    },
  },
];

export const adaptedOffers = serverOffers.slice().map((offer) => offer);
export const adaptedReviews = serverReviews.slice().map((comment) => comment);

export const reviewData = {
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  rating: 4,
};

export const userData = {
  email: `Oliver.conner@gmail.com`,
  password: `12345678`,
};

export const authInfo = {
  avatar_url: `img/1.png`,
  email: `Oliver.conner@gmail.com`,
  id: 1,
  is_pro: false,
  name: `Oliver.conner`,
};

export const enteredInfo = {
  login: `test@mail.ru`,
  password: `qwerty12`,
};
