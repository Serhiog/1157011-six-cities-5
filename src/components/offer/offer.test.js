import React from "react";
import renderer from "react-test-renderer";
import {Offer} from "./offer";
import {serverOffers} from "../../mocks/data";
import {MemoryRouter} from "react-router-dom";
jest.mock(`../review-list/review-list`, () => `ReviewList`);
jest.mock(`../map/map`, () => `Map`);
jest.mock(`../offers-list/offer-list`, () => `OfferList`);
jest.mock(`../favorite-btn/favorite-btn`, () => `FavoriteBtn`);

it(`Should Offer render correctly`, () => {
  const tree = renderer
    .create(
        <MemoryRouter>
          <Offer
            offer={serverOffers[0]}
            nearbyOffers={serverOffers}
            actualOffer={serverOffers[0]}
            getReviews={() => {}}
            authorizationStatus={`AUTH`}
            isLogged={true}
            email={`email@email.com`}
            goToFavorites={() => {}}
            getNearbyOffers={serverOffers}
          />
        </MemoryRouter>,
        {
          createNodeMock: () => {
            return {};
          },
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
