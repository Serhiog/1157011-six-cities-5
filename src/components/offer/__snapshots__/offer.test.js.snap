// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Offer render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height={41}
              src="/img/logo.svg"
              width={81}
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign In
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Place image"
              className="property__image"
              src="img/1.png"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Place image"
              className="property__image"
              src="img/2.png"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious studio at great location
            </h1>
            <FavoriteBtn
              auth={true}
              classCard="property"
              offer={
                Object {
                  "bedrooms": 3,
                  "city": Object {
                    "location": Object {
                      "latitude": 52.370216,
                      "longitude": 4.895168,
                      "zoom": 10,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                  "goods": Array [
                    "Heating",
                    "Kitchen",
                    "Cable TV",
                    "Washing machine",
                    "Coffee machine",
                    "Dishwasher",
                  ],
                  "host": Object {
                    "avatar_url": "img/1.png",
                    "id": 3,
                    "is_pro": true,
                    "name": "Angelina",
                  },
                  "id": 1,
                  "images": Array [
                    "img/1.png",
                    "img/2.png",
                  ],
                  "is_favorite": false,
                  "is_premium": false,
                  "location": Object {
                    "latitude": 52.35514938496378,
                    "longitude": 4.673877537499948,
                    "zoom": 8,
                  },
                  "max_adults": 4,
                  "preview_image": "img/1.png",
                  "price": 120,
                  "rating": 4.8,
                  "title": "Beautiful & luxurious studio at great location",
                  "type": "apartment",
                }
              }
            />
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.8
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              4
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              120
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Cable TV
              </li>
              <li
                className="property__inside-item"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
              >
                Coffee machine
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="/img/1.png"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <ReviewList
              offerId={1}
            />
          </section>
        </div>
      </div>
      <Map
        cityCoordinates={
          Array [
            52.370216,
            4.895168,
          ]
        }
        classMap="property__map"
        forOffer={true}
        mainOffer={
          Object {
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Amsterdam",
            },
            "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
            "goods": Array [
              "Heating",
              "Kitchen",
              "Cable TV",
              "Washing machine",
              "Coffee machine",
              "Dishwasher",
            ],
            "host": Object {
              "avatar_url": "img/1.png",
              "id": 3,
              "is_pro": true,
              "name": "Angelina",
            },
            "id": 1,
            "images": Array [
              "img/1.png",
              "img/2.png",
            ],
            "is_favorite": false,
            "is_premium": false,
            "location": Object {
              "latitude": 52.35514938496378,
              "longitude": 4.673877537499948,
              "zoom": 8,
            },
            "max_adults": 4,
            "preview_image": "img/1.png",
            "price": 120,
            "rating": 4.8,
            "title": "Beautiful & luxurious studio at great location",
            "type": "apartment",
          }
        }
        mapZoom={10}
        offers={
          Array [
            Object {
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatar_url": "img/1.png",
                "id": 3,
                "is_pro": true,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "img/1.png",
                "img/2.png",
              ],
              "is_favorite": false,
              "is_premium": false,
              "location": Object {
                "latitude": 52.35514938496378,
                "longitude": 4.673877537499948,
                "zoom": 8,
              },
              "max_adults": 4,
              "preview_image": "img/1.png",
              "price": 120,
              "rating": 4.8,
              "title": "Beautiful & luxurious studio at great location",
              "type": "apartment",
            },
            Object {
              "bedrooms": 4,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatar_url": "img/1.png",
                "id": 3,
                "is_pro": true,
                "name": "Angelina",
              },
              "id": 2,
              "images": Array [
                "img/1.png",
                "img/2.png",
              ],
              "is_favorite": false,
              "is_premium": true,
              "location": Object {
                "latitude": 52.35514938496378,
                "longitude": 4.673877537499948,
                "zoom": 8,
              },
              "max_adults": 4,
              "preview_image": "img/1.png",
              "price": 100,
              "rating": 4.8,
              "title": "Beautiful & luxurious studio at great location",
              "type": "apartment",
            },
          ]
        }
        selectedCity={
          Object {
            "location": Object {
              "latitude": 52.370216,
              "longitude": 4.895168,
              "zoom": 10,
            },
            "name": "Amsterdam",
          }
        }
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <OfferList
          classCard="near-places__card"
          classImageWrapper="near-places__image-wrapper"
          classList="near-places__list"
          forFavorites={true}
          nearby={true}
          offers={
            Array [
              Object {
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Amsterdam",
                },
                "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "host": Object {
                  "avatar_url": "img/1.png",
                  "id": 3,
                  "is_pro": true,
                  "name": "Angelina",
                },
                "id": 1,
                "images": Array [
                  "img/1.png",
                  "img/2.png",
                ],
                "is_favorite": false,
                "is_premium": false,
                "location": Object {
                  "latitude": 52.35514938496378,
                  "longitude": 4.673877537499948,
                  "zoom": 8,
                },
                "max_adults": 4,
                "preview_image": "img/1.png",
                "price": 120,
                "rating": 4.8,
                "title": "Beautiful & luxurious studio at great location",
                "type": "apartment",
              },
              Object {
                "bedrooms": 4,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Amsterdam",
                },
                "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                "goods": Array [
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                  "Washing machine",
                  "Coffee machine",
                  "Dishwasher",
                ],
                "host": Object {
                  "avatar_url": "img/1.png",
                  "id": 3,
                  "is_pro": true,
                  "name": "Angelina",
                },
                "id": 2,
                "images": Array [
                  "img/1.png",
                  "img/2.png",
                ],
                "is_favorite": false,
                "is_premium": true,
                "location": Object {
                  "latitude": 52.35514938496378,
                  "longitude": 4.673877537499948,
                  "zoom": 8,
                },
                "max_adults": 4,
                "preview_image": "img/1.png",
                "price": 100,
                "rating": 4.8,
                "title": "Beautiful & luxurious studio at great location",
                "type": "apartment",
              },
            ]
          }
        />
      </section>
    </div>
  </main>
</div>
`;
